RFCs/FS-1061-Allow-intrinsic-type-extensions-for-provided-types.md

```diff
@@ -79,19 +79,101 @@ This would result in the following warning:
+
+    override x.DidReceiveMemoryWarning() = 
+        base.DidReceiveMemoryWarning()
+        
+    override x.ViewDidLoad() =
+        x.DoNotPressAction <- Some (fun _ -> x.View.BackgroundColor <- UIColor.Purple)
+        base.ViewDidLoad()
+```
+
+This has several drawbacks that makes the user facing code more complex and unweidy than it should be with additions mentioned in the RFC.
+
+## Attributes have to be defined on outward facing type
+
+Many frameworks require the generation of attributes to be applied to generated type, Xamarin.iOS is such an example.  The `Register` attribute has to be applied to `ViewControllers` to supply the mapping between a user class and the storyboard name.  As attributes do not propagate with inheritance you cannot generate them on the base or abstract type.  The best option available to a type provider designer is to provide a helper to be generated that the consumer has to learn how to apply, of course this can be detailed in documentation but its an example of unnecessary work.  You can see this in the iOS UI provider above:
+
+```
+[<Register(container.myViewControllerBase.CustomClass)>]
+type ViewController(handle : IntPtr) =  
+```
+
+## Inheritance required
+
+The generated type **has** to be sub-classed to apply method overloads which are required by certain framework, iOS being the example again. Xamarin.iOS requires that the users custom `ViewController` to inherit from `UIViewController` or a derivative, these base types had a multitude of methods that can be overridden such as `ViewDidLoad` and `DidReceiveMemoryWarning` in the example above.  This means any constructors present on the type must also be detailed using the inherit syntax:
+
+```fsharp
+type ViewController(handle : IntPtr) =  
+    inherit container.myViewControllerBase(handle)
+```
+
+In the case of multiple constructors being required the record syntax must be used which starts to become cumbersome and ugly to define.  An example of this can be seen if you take this example from the Android ViewUI type provider:
+
+```fsharp
+type container2 = Xamarin.Android.ViewUI<"Test.axml"> 
+
+type MyView =
+    inherit container2
+    new() as x =
+        { inherit container2()} then x.Initialise()
+    new(context) as x = 
+        { inherit container2(context) } then x.Initialise()
+    new(context:Context, attr:Android.Util.IAttributeSet) as x =
+        { inherit container2(context, attr) } then x.Initialise()
+```
+
+All of the duplicated constructor propagation can be avoided by allowing intrinsic type extensions  to occur on generative type providers.
+
+Method overloads are applied in the usual manner but if they could be applied as part of the 
+```fsharp
+    override x.ViewDidLoad() =
+        x.DoNotPressAction <- Some (fun _ -> x.View.BackgroundColor <- UIColor.Purple)
+        base.ViewDidLoad()
+```
+
+If this RFC was implemented the iOS type provider would look like this instead:
```

Comment:
Nit: is this supposed to say "Type provider" or "usage of the iOS type provider"?

Reply:
the latter

Reply:
updated text